xarray/core/combine.py

```diff
@@ -749,6 +759,13 @@ def combine_by_coords(
         precipitation  (y, x) float64 0.4376 0.8918 0.9637 ... 0.5684 0.01879 0.6176
     """
 
+    # If a set of unnamed data arrays is provided, these arrays are assumed to belong
+    # to the same variable and should be combined.
+    if _all_unnamed_data_arrays(data_objects):
+        datasets = [Dataset({"_": data_array}) for data_array in data_objects]
```

Comment:
Should we be using `._to_temp_dataset` here? See @dcherian 's [comment here](https://github.com/pydata/xarray/issues/3248#issuecomment-531825386).